packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -167,7 +167,18 @@ const createHistoryService = ({ strapi }: { strapi: Core.Strapi }) => {
             : { documentId: context.params.documentId, locale: context.params?.locale };
 
         const defaultLocale = await getDefaultLocale();
-        const locale = documentContext.locale || defaultLocale;
+
+        let locale = defaultLocale;
+        if (documentContext.locale) {
+          if (Array.isArray(documentContext.locale)) {
+            // TODO calls picked from the middleware could contain an array of
+            // locales. This is incompatible with our call to findOne below.
+
+            return next();
+          }
+
+          locale = documentContext.locale;
+        }
```

Comment:
Could we avoid creating a mutable variable for locale? Wouldn't this be enough?

```suggestion
if (Array.isArray(documentContext.locale)) {
  // TODO calls picked from the middleware could contain an array of
  // locales. This is incompatible with our call to findOne below.

  return next();
}

const locale = documentContext.locale || defaultLocale;
```